### ISSUE TITLE
import/named error for JSDoc typedef

### ISSUE THREAD

**Issue by matt-tingen** [import-js/eslint-plugin-import#1381]
In TypeScript with the `allowJs` flag, types defined in a JSDoc can be imported in `.ts` files, but the `import/named` rule is not aware of the implicit export from the JSDoc and thus marks the import as invalid.

Example:
```js
// Foo.js
/** @typedef {keyof Foo} FooKey */
class Foo { /* ... */ }
export default Foo;
```

```ts
// Bar.ts
import Foo, { FooKey } from './Foo';
```

This yields `error  FooKey not found in './Foo'  import/named`.

It may be possible to use [tsdoc](https://github.com/microsoft/tsdoc) to extract the type definitions.

Related: #1282

**Comment by ljharb** [comment]
oof, this seems very complex. i'd want to enable this by a setting only, not by default.

**Comment by matt-tingen** [comment]
Absolutely. Honestly, I'm not totally sure how worth it it would be to implement. Personally, I just disabled `import/named` for `.tsx?` files in our project since TS performs the check, anyway.

**Comment by JounQin** [comment]
You can use

```ts
import Foo, { type FooKey } from './Foo';
```

It won't error now.
